src/core/processing/processingalgorithmparametersmodel.cpp

```diff
@@ -283,6 +285,20 @@ QVariant ProcessingAlgorithmParametersModelBase::data( const QModelIndex &index,
         const QgsProcessingParameterEnum *parameterEnum = dynamic_cast<const QgsProcessingParameterEnum *>( mParameters.at( index.row() ) );
         configuration["options"] = parameterEnum->options();
       }
+      else if ( mParameters.at( index.row() )->type() == QStringLiteral( "source" ) )
+      {
+        QMap<QString, QgsMapLayer *> map = QgsProject::instance()->mapLayers();
+        QStringList list;
```

Comment:
Instead of a string list of layer names here we should return a QVariantMap with keys being the layer _ID_ and the value being the layer name.  Referring to a layer by its ID is more resilient.

Reply:
Thanks @nirvn for the advices, indeed it is more resilient to search by ids